ibm/resource_ibm_cis_healthcheck.go

```diff
@@ -2,331 +2,364 @@ package ibm
+		if timeout, ok := d.GetOk(cisGLBHealthCheckTimeout); ok {
+			opt.SetTimeout(int64(timeout.(int)))
 		}
-		if retries, ok := d.GetOk("retries"); ok {
-			monitorUpdate.Retries = retries.(int)
+		if retries, ok := d.GetOk(cisGLBHealthCheckRetries); ok {
+			opt.SetRetries(int64(retries.(int)))
 		}
-		if interval, ok := d.GetOk("interval"); ok {
-			monitorUpdate.Interval = interval.(int)
+		if interval, ok := d.GetOk(cisGLBHealthCheckInterval); ok {
+			opt.SetInterval(int64(interval.(int)))
 		}
-		if follow_redirects, ok := d.GetOk("follow_redirects"); ok {
-			monitorUpdate.FollowRedirects = follow_redirects.(bool)
+		if followRedirects, ok := d.GetOk(cisGLBHealthCheckFollowRedirects); ok {
+			opt.SetFollowRedirects(followRedirects.(bool))
 		}
-		if allow_insecure, ok := d.GetOk("allow_insecure"); ok {
-			monitorUpdate.AllowInsecure = allow_insecure.(bool)
+		if allowInsecure, ok := d.GetOk(cisGLBHealthCheckAllowInsecure); ok {
+			opt.SetAllowInsecure(allowInsecure.(bool))
 		}
-		if port, ok := d.GetOk("port"); ok {
-			monitorUpdate.Port = port.(int)
+		if port, ok := d.GetOk(cisGLBHealthCheckPort); ok {
+			opt.SetPort(int64(port.(int)))
 		}
-		_, err = cisClient.Monitors().UpdateMonitor(cisId, monitorId, monitorUpdate)
+		result, resp, err := sess.EditLoadBalancerMonitor(opt)
 		if err != nil {
-			log.Printf("[WARN] Error getting zone during MonitorUpdate %v \n", err)
+			log.Printf("Error updating global load balancer health check detail: %s", resp)
 			return err
 		}
+		log.Printf("Monitor update succesful : %s", *result.Result.ID)
 	}
-	return resourceCIShealthCheckRead(d, meta)
+
+	return resourceCISHealthCheckRead(d, meta)
 }
 
-func resourceCIShealthCheckDelete(d *schema.ResourceData, meta interface{}) error {
-	cisClient, err := meta.(ClientSession).CisAPI()
+func resourceCISHealthCheckDelete(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).CisGLBHealthCheckClientSession()
 	if err != nil {
 		return err
 	}
-	monitorId, cisId, err := convertTftoCisTwoVar(d.Id())
+
+	monitorID, crn, err := convertTftoCisTwoVar(d.Id())
 	if err != nil {
 		return err
 	}
-	var monitor *v1.Monitor
-	emptyMonitor := new(v1.Monitor)
-	monitor, err = cisClient.Monitors().GetMonitor(cisId, monitorId)
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewDeleteLoadBalancerMonitorOptions(monitorID)
+
+	result, resp, err := sess.DeleteLoadBalancerMonitor(opt)
 	if err != nil {
-		if checkCisMonitorDeleted(d, meta, err, monitor) {
-			d.SetId("")
-			return nil
-		}
-		log.Printf("[WARN] Error getting zone during MonitorRead %v\n", err)
+		log.Printf("Error deleting global load balancer health check detail: %s", resp)
 		return err
 	}
-
-	monitorObj := *monitor
-	if !reflect.DeepEqual(emptyMonitor, monitorObj) {
-		err = cisClient.Monitors().DeleteMonitor(cisId, monitorId)
-		if err != nil {
-			log.Printf("[WARN] DeleteMonitor Failed %s\n", err)
-			return err
-		}
-	}
-
-	d.SetId("")
```

Comment:
set id to nil before returning
revert back `d.SetId("")`

Reply:
checkCisMonitorDeleted() is a dead code since I have added Exists() method

Reply:
d.SetId("") id isnt set to nil in before returning in destroy method

Reply:
we have Exists() method which does what you are trying to say here. Thats why I am saying checkDelete() method is a dead code here. it Never executes till the Exists() method gets removed..
please understand it.I tested it multiple time with different scenario.
 other wise test my code yourself. if you are saying that checkDelete() method still executed on particular scenario, please let me know the scenario.

Reply:
okay..